src/os/ogc/ogcservice.js

```diff
@@ -0,0 +1,399 @@
+   */
+  getTags(feature) {
+    var tags;
+    var props = this.properties_ ? os.settings.get(this.properties_) : null;
+    if (props) {
+      var values = [];
+      for (var i = 0; i < props.length; i++) {
+        var value = feature.get(props[i]);
+        if (value) {
+          values.push(value);
+        }
+      }
+
+      tags = values.join(', ');
+    }
+    return tags;
+  }
+
+  /**
+   * Handle area being loaded from the server.
+   * @param {!T} datum The requested area
+   * @param {!ol.Feature} feature The area border feature
+   * @return {!ol.Feature}
+   * @template T
+   */
+  populateFeature(datum, feature) {
+    var tags = this.getTags(feature);
+
+    feature.set('title', this.getLabel(feature));
+    feature.set(this.nameProperty_, undefined);
+    feature.set('tags', 'Borders' + (tags ? (', ' + tags) : ''));
+
+    this.fix_(feature);
+
+    return feature;
+  }
+
+  /**
+   * @param {!string} id
+   */
+  setServiceId(id) {
+    this.serviceId_ = id;
+  }
+
+  /**
+   * @param {function(!Array<!ol.Feature>): ?} resolve The Promise resolve function
+   * @param {function(*): ?} reject The goog.Promise reject function
+   * @private
+   */
+  loadAll_(resolve, reject) {
+    if (this.isConfigured()) {
+      var qd = os.ogc.getDefaultWfsParams();
+      qd.set('typename', this.typename_);
+      qd.set('propertyname', 'ID,' + this.nameProperty_);
+      qd.set('outputformat', 'text/xml; subtype=gml/3.1.1');
+
+      var uri = new Uri(this.url_);
+      uri.setQueryData(qd);
+
+      var request = new Request(uri);
+      request.setHeader('Accept', 'application/xml');
+      request.setValidator(os.ogc.getException);
+      request.setDataFormatter(new WFSFormatter());
+
+      request.listenOnce(goog.net.EventType.SUCCESS, goog.partial(this.onAllLoaded_, resolve, reject), false, this);
+      request.listenOnce(goog.net.EventType.ERROR, function(event) {
+        this.onError(event, 'Failed to load data list.', false, reject);
+      }.bind(this), false, this);
+
+      request.load();
+    } else {
+      reject('Service is not configured.');
+    }
+  }
+
+  /**
+   * Handle data being loaded from the server.
+   * @param {function(!Array<!T>)} resolve Promise resolution function
+   * @param {function(*)} reject Promise rejection function
+   * @param {goog.events.Event} event
+   * @template T
+   * @private
+   */
+  onAllLoaded_(resolve, reject, event) {
+    var request = /** @type {Request} */ (event.target);
+    var response = /** @type {string|undefined} */ (request.getResponse());
+    goog.dispose(request);
+
+    var list = [];
+    if (response) {
+      var parser = new GMLParser();
+      parser.setSource(response);
+
+      while (parser.hasNext()) {
+        var feature = parser.parseNext();
+        if (feature) {
+          list.push(feature);
+        }
+      }
+    }
+
+    if (list && list.length > 0) {
+      list.sort(goog.partial(osOlFeature.fieldSort, this.nameProperty_));
```

Comment:
`goog.partial` calls can be replaced with `fn.bind(undefined, args)`. We'll replace them all some day,  but they're generally not preferred anymore.